src/ipa_pc/mod.rs

```diff
@@ -672,21 +725,45 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
             let hiding_bound = labeled_polynomial.hiding_bound();
             let commitment = labeled_commitment.commitment();
 
-            combined_polynomial += (cur_challenge, polynomial);
-            combined_commitment_proj += &commitment.comm.mul(cur_challenge);
+            let p_len = polynomial.coeffs.len();
+
+            if p_len > key_len {
+
+                let mut polynomial_lc = Polynomial::zero();
+                for i in 0..p_len / key_len + if p_len % key_len != 0 { 1 } else { 0 } {
+                    let is = i * key_len;
+                    let poly_single = Polynomial::from_coefficients_slice(&polynomial.coeffs[i * key_len..core::cmp::min((i + 1) * key_len, p_len)]);
+                    polynomial_lc += (point.pow(&[is as u64]), &poly_single);
+                }
```

Comment:
As in my comment above: Let us use the Horner scheme to evaluate the polynomial_lc.

Reply:
As I understand horner scheme should be implemented in ginger-lib, not in this repo, right?